.github/workflows/ci.yml

```diff
@@ -99,6 +99,78 @@ jobs:
           name: test-results
           path: 'up-cpp/build/test/results/*.xml'
 
+# NOTE: Run dynamic analysis in unit tests
+  valgrind:
+    name: Run up-cpp tests with Valgrind
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - name: Get build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-artifacts
+          path: up-cpp/build
+
+      - name: Install Valgrind
+        run: sudo apt-get install -y valgrind
+
+      - name: Run Valgrind Memcheck
+        run: |
+          cd up-cpp/build
+          chmod +x bin/*
+          echo "Running Memcheck..."
+          valgrind --leak-check=full \
+                   --show-leak-kinds=all \
+                   --track-origins=yes \
+                   --verbose \
+                   --log-file=valgrind-memcheck.log \
+                   --gen-suppressions=all \
+                   ctest
```

Comment:
I'm a little concerned that `valgrind ... ctest` will result in valgrind checking only the `ctest` binary and not the actual test applications. Have you verified that this _does_ catch a leak in a test if one is introduced intentionally?

Reply:
I see, that's a great  point. Valgrind documentation tells to set --trace-children=yes to trace child processes.

I will try this by setting --trace-children flag

Reply:
You should be able to verify it is working by adding an unmatched `auto x = new int(3);` or something like that to a source file where it will get run by a test.

Reply:
sounds good, it should be relatively straight to setup valgrind locally and test this :)